From b0a09f044f89d7f54d7abdcb47f91f696392c8b6 Mon Sep 17 00:00:00 2001
From: Bernhard Rosenkränzer <Bernhard.Rosenkranzer@linaro.org>
Date: Sun, 27 Aug 2017 15:36:57 +0200
Subject: [PATCH] Don't crash when displaying the " key

When displaying any key selection that has a " key, AOSP 8.0.0_r4's
LatinIME crashes.

The problem is that XML escapes are handled too early:
latin:keySpec="&quot;" becomes latin:keySpec=""" before it's fed to
an XML parser, causing a syntax error.

https://issuetracker.google.com/issues/65090157

Change-Id: I7b52fc62c874075d572a631c75e87caad1d365c0
Signed-off-by: Bernhard Rosenkränzer <Bernhard.Rosenkranzer@linaro.org>
---
 java/res/xml/rowkeys_symbols3.xml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/java/res/xml/rowkeys_symbols3.xml b/java/res/xml/rowkeys_symbols3.xml
index 8093081..dfc8bfb 100644
--- a/java/res/xml/rowkeys_symbols3.xml
+++ b/java/res/xml/rowkeys_symbols3.xml
@@ -37,7 +37,7 @@
         </case>
         <default>
             <Key
-                latin:keySpec="&quot;"
+                latin:keySpec="\&quot;"
                 latin:moreKeys="!text/morekeys_double_quote" />
             <Key
                 latin:keySpec="\'"
--
libgit2 0.25.0

